client/src/main/java/io/pravega/client/stream/impl/LargeEventWriter.java

```diff
@@ -0,0 +1,252 @@
+
+        CreateTransientSegment createSegment = new CreateTransientSegment(requestId,
+                writerId,
+                parentSegment.getScopedName(),
+                token);
+
+        SegmentCreated created = transformSegmentCreated(getThrowingException(client.sendRequest(requestId, createSegment)),
+                                                         parentSegment.getScopedName());
+        requestId = client.getFlow().getNextSequenceNumber();
+        SetupAppend setup = new SetupAppend(requestId, writerId, created.getSegment(), token);
+
+        AppendSetup appendSetup = transformAppendSetup(getThrowingException(client.sendRequest(requestId, setup)),
+                                                       created.getSegment());
+
+        if (appendSetup.getLastEventNumber() != 0) {
+            throw new IllegalStateException(
+                    "Server indicates that transient segment was already written to: " + created.getSegment());
+        }
+
+        long expectedOffset = 0;
+        val futures = new ArrayList<CompletableFuture<Reply>>();
+        for (int i = 0; i < payloads.size(); i++) {
+            requestId = client.getFlow().getNextSequenceNumber();
+            ByteBuf payload = payloads.get(i);
+            val request = new ConditionalBlockEnd(writerId,
+                    i,
+                    expectedOffset,
+                    Unpooled.wrappedBuffer(payload),
+                    requestId);
+            expectedOffset += payload.readableBytes();
+            val reply = client.sendRequest(requestId, request);
+            failFast(futures, created.getSegment());
+            futures.add(reply);
+        }
+        for (int i = 0; i < futures.size(); i++) {
+            transformDataAppended(getThrowingException(futures.get(i)), created.getSegment());
+        }
+        requestId = client.getFlow().getNextSequenceNumber();
+
+        MergeSegments merge = new MergeSegments(requestId, created.getSegment(), parentSegment.getScopedName(), token);
+
+        transformSegmentMerged(getThrowingException(client.sendRequest(requestId, merge)), created.getSegment());
+    }
+
+    // Trick to fail fast if any of the futures have completed.
+    private void failFast(ArrayList<CompletableFuture<Reply>> futures, String segmentId) throws TokenExpiredException,
+            NoSuchSegmentException, AuthenticationException, SegmentSealedException, ConnectionFailedException {
+        for (CompletableFuture<Reply> future : futures) {
+            if (!future.isDone()) {
+                break;
+            } else {
+                transformDataAppended(getThrowingException(future), segmentId);
+            }
+        }
+    }
+
+    private SegmentCreated transformSegmentCreated(Reply reply, String segmentId) throws TokenExpiredException,
+            NoSuchSegmentException, AuthenticationException, SegmentSealedException, ConnectionFailedException {
+        if (reply instanceof SegmentCreated) {
```

Comment:
Can we use `reply.getType()` instead of type checking? `RawClient.sendRequest` returns a Future containing a `Reply & WireCommand` so we can definitely extract the `getType()` without casting or type checking.